### DISCUSSION TITLE
Difference between edge_index and edge_label_index

### DISCUSSION THREAD

**Discussion by Jeriousman** [pyg-team/pytorch_geometric#3975] [Q&A] [answered]
Literally what is difference between edge_index and edge_label_index?

I guess it has something to do with negative sampling but I wish I can get some firm detailed information with some examples.

**Comment by rusty1s** [comment, marked as answer]
Sorry for the confusion.

* `edge_index` denotes the edges that are used for message passing inside a GNN
* `edge_label_index` denote the pairs of nodes that are used for supervision/loss computation. For example, we try to predict new pairs of links in `edge_label_index` after propagating information and learning node embeddings based on `edge_index`. In addition, as you correctly pointed out, `edge_label_index` contains both positive and negative pairs of examples.

**Reply by Jeriousman** [reply]
Thank you for your clear answer!
